Various improvements re character encoding and testing

asmundg opened the pull request:
These are mostly improvements I needed myself, but they're relevant for #7, #9 and #10.

Commit:
Make code py3 compatible

As a side-effect, this makes us use unicode, which will be useful for
non-ascii characters in aythor and story names down the line.

Commit:
Handle non-ascii author names

Commit:
Add some initial unit tests

Just to prove that my initial changes work.

Commit:
Handle encoding correctly when piping

Python doesn't let us know the terminal encoding if we're piping (which
may or may not be correct, depending on who you ask). So we need some
extra handling.

Incidentally, handling the output differently also makes is easier to
test that we're getting the right output.

I've only implemented this for show_stories thus far.

Commit:
Start separating concerns

Fetching and displaying information are two distinct tasks, one of which
is functional.

Commit:
Add proper output processing for scrum function

Also basic testing and separation of concerns.

Commit:
Test against python 3.2 as well, just to be sure

Commit:
Allow finished and delivered in scrum

Seeing these is useful in some workflows,

weldyss commented:
I've tested here and the story title stills bad. I received the unicode error on `pivotal_tools show stories` here

jtushman commented:
yay -- pull request -- I will review shortly

jtushman commented:
(and sorry I did not see this earlier)

ropez committed:
Use the binary response content instead of unicode

Apparently, ElementTree likes to do it's own decoding. It raises a
UnicodeError when given the decoded response.text, but it works fine
when given the raw response.content.

asmundg committed:
Merge pull request #1 from ropez/master

Use the binary response content instead of unicode

cj13579 commented:
So I know that this pull hasn't been merged or accepted yet but I just downloaded the fork and it makes something that doesn't work, work! You should probably merge this 😄 👍
